### PR TITLE
User able to post and view other's posts

### PR DESCRIPTION
**Pull request by craig-a-lee**

### CODE REVIEW

**Comment by craig-a-lee** [comment]
Demo: https://imgur.com/a/AS0TXG2

File: Wellthier/Managers/HealthKitSharedManager.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+//  HealthKitSharedManager.h
+//  Wellthier
+//
+//  Created by Craig Lee on 7/20/22.
+//
+
+#import <Foundation/Foundation.h>
+#import "HealthKit/HealthKit.h"
```
Comment (vijaykanse) [inline Wellthier/Managers/HealthKitSharedManager.h]: #import <HealthKit/HealthKit.h>

File: Wellthier/Managers/HealthKitSharedManager.m [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        commonManager.healthStore = [[HKHealthStore alloc] init];
+    });
+    return commonManager;
+}
+
+- (void) requestAuthorization {
+    if (![HKHealthStore isHealthDataAvailable]) {
+        return;
+    }
+
```
Comment (vijaykanse) [inline Wellthier/Managers/HealthKitSharedManager.m]: Extra line

File: Wellthier/Managers/HealthKitSharedManager.m [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    return commonManager;
+}
+
+- (void) requestAuthorization {
+    if (![HKHealthStore isHealthDataAvailable]) {
+        return;
+    }
+    
+    NSSet *readTypes = [[NSSet alloc] initWithArray:@[[HKObjectType workoutType], [HKSeriesType activitySummaryType], [HKSeriesType workoutType]]];
+
```
Comment (vijaykanse) [inline Wellthier/Managers/HealthKitSharedManager.m]: Extra line

File: Wellthier/Managers/HealthKitSharedManager.m [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+- (HKWorkout *) getLatestWorkout {
+    HKObjectType *type = [HKObjectType workoutType];
+    __block HKWorkout *latestWorkout = nil;
+    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey: HKSampleSortIdentifierStartDate ascending:false];
+    HKQuery *query = [[HKSampleQuery alloc] initWithSampleType:type predicate:nil limit:HKObjectQueryNoLimit sortDescriptors:@[sortDescriptor] resultsHandler:^(HKSampleQuery * _Nonnull query, NSArray<__kindof HKSample *> * _Nullable results, NSError * _Nullable error) {
+        if (results.count > 0) {
+            latestWorkout = results[0];
+
```
Comment (vijaykanse) [inline Wellthier/Managers/HealthKitSharedManager.m]: Extra line

File: Wellthier/View Controllers/ComposeViewController.m [hunk 5]
```diff
@@ -7,6 +7,8 @@
 
 #import "ComposeViewController.h"
 #import "Parse/Parse.h"
+#import "HealthKit/HealthKit.h"
```
Comment (vijaykanse) [inline Wellthier/View Controllers/ComposeViewController.m]: #import <HealthKit/HealthKit.h>

File: Wellthier/View Controllers/FeedViewController.m [hunk 6]
```diff
@@ -6,15 +6,19 @@
 //
 
 #import "FeedViewController.h"
+#import "HealthKitSharedManager.h"
 
 @interface FeedViewController ()
 
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
```
Comment (vijaykanse) [inline Wellthier/View Controllers/FeedViewController.m]: order for property attributes

**Review by vijaykanse — APPROVED** [review]

**Pull request merged**